dbms/src/Flash/Planner/plans/PhysicalJoin.cpp

```diff
@@ -142,13 +175,24 @@ PhysicalPlanNodePtr PhysicalJoin::build(
     size_t max_block_size_for_cross_join = settings.max_block_size;
     fiu_do_on(FailPoints::minimum_block_size_for_cross_join, { max_block_size_for_cross_join = 1; });
 
+    size_t shuffle_partition_num = 0;
+    if (fine_grained_shuffle.enable())
```

Comment:
Maybe we can use `Join::join_build_concurrency directly`?
Because join_build_concurrency  is the build side streams num.
In this case, the join_build_concurrency also is the ReceiverSourceNum.

Reply:
Oh, that's not the same concept:). join_build_concurrency probably equals fine_grained_stream_count, shuffle_partition_num is the number of total TiFlash servers.

Reply:
o........yes... My mistake...